src/compiler/core.ts

```diff
@@ -1532,7 +1532,6 @@ namespace ts {
         }
         if (path.charCodeAt(1) === CharacterCodes.colon) {
             if (path.charCodeAt(2) === CharacterCodes.slash) return 3;
-            return 2;

```

Comment:
Should it still return 2 if `path.length === 2` to handle input like `C:`?

Reply:
seems like a good suggestion. 
we should add some unit tests for this function, but that can be a different change.

Reply:
`C:` does not look like a valid path to me, so I'd rather say no.

Maybe I'm just missing some use cases.

Reply:
`c:` is a valid folder name on windows. so we should allow it.

Reply:
Well, I've made a little research. Here's what I've found.

1. `c:` is NOT a valid folder name on Windows. `:` is not a valid character for a folder name.
2. In command line `cd c:` does nothing, `dir c:` displays the content of a current directory. So it seems like `c:` is not a path, but rather a volume label, which is not handled by Windows the same way as a regular path.
3. Node's `path.isAbsolutePath` returns `true` for both `c:/` and `c:\\`, but `false` for `c:`.
4. Node's `path.resolve` returns current directory for `c:`.

Based on this behaviour a cannot see how `c:` is a valid path.

Maybe you can provide some code to illustrate your point?

Reply:
@mhegazy Take a look, please. ☝️

Reply:
fair enough. can you update the test baselines

Reply:
I did, and something went wrong : [

I've run exactly

```
jake
jake runtests-parallel
jake baseline-accept
```

Now there are tons of changed and new files in the PR. What should I've done differently?

Reply:
@mhegazy Can you take a look please?